tests/unit/lms/resources/_js_config/__init___test.py

```diff
@@ -286,98 +287,81 @@ def authToken(self, config):
                 },
             },
-        }
+        )
 
-    @pytest.mark.usefixtures("with_sections_on", "learner_canvas_user_id")
-    def test_it_adds_learner_canvas_user_id_for_SpeedGrader_launches(self, sync):
-        assert sync["data"]["learner"] == {
-            "canvas_user_id": "test_learner_canvas_user_id",
+        assert js_config.asdict()["api"]["sync"] == {
+            "authUrl": "route:auth_route",
+            "path": "route:sync_route",
+            "data": grouping_plugin.get_grouping_sync_config.return_value,
         }
 
-    def test_its_None_if_section_and_groups_arent_enabled(self, sync):
-        assert sync is None
+    def test_it_does_not_add_config_if_grouping_is_course(self, js_config):
+        context.grouping_type = Grouping.Type.COURSE
 
-    @pytest.fixture
-    def sync(self, config, js_config):
-        # Call enable_lti_launch_mode() so that the api.sync section gets
-        # inserted into the config.
         js_config.enable_lti_launch_mode()
 
-        return config["api"]["sync"]
+        assert not js_config.asdict()["api"]["sync"]
 
-    @pytest.fixture
-    def learner_canvas_user_id(self, pyramid_request):
-        pyramid_request.params["learner_canvas_user_id"] = "test_learner_canvas_user_id"
+    def test_it_does_not_add_config_if_no_sync_url(self, js_config, grouping_plugin):
+        context.grouping_type = Grouping.Type.GROUP
+        grouping_plugin.sync_route = None
+
+        js_config.enable_lti_launch_mode()
+
+        assert not js_config.asdict()["api"]["sync"]
 
     @pytest.fixture
-    def blackboard_group_launch(self, context, pyramid_request):
-        context.grouping_type = Grouping.Type.GROUP
-        pyramid_request.product.family = Product.Family.BLACKBOARD
+    def grouping_plugin(self):
+        grouping_plugin = create_autospec(GroupingServicePlugin, spec_set=True)
+        grouping_plugin.auth_route = "auth_route"
+        grouping_plugin.sync_route = "sync_route"
+
+        return grouping_plugin
 
     @pytest.fixture
-    def pyramid_request(self, pyramid_request):
+    def pyramid_request(self, pyramid_request, grouping_plugin):
         pyramid_request.params.clear()
         pyramid_request.params.update(
             {
                 "context_id": "test_context_id",
                 "custom_canvas_course_id": "test_custom_canvas_course_id",
                 "resource_link_id": "test_resource_link_id",
                 "tool_consumer_instance_guid": "test_tool_consumer_instance_guid",
-                "foo": "bar",  # This item should be missing from group_info.
             }
         )
-        return pyramid_request
+        pyramid_request.product = Product.from_request(pyramid_request)
+        pyramid_request.product.plugin.grouping_service = grouping_plugin
+
+        with patch.object(pyramid_request, "route_url") as route_url:
+            route_url.side_effect = lambda route_name, **kwargs: f"route:{route_name}"
+            yield pyramid_request
```

Comment:
If this was all in the plugin it would be much simpler to test. It would also cut down the giagnormous file here.